src/libDirectoryService/PoWProcessing.cpp

```diff
@@ -90,30 +90,61 @@ bool DirectoryService::VerifyPoWSubmission(
     rand1 = m_mediator.m_dsBlockRand;
     rand2 = m_mediator.m_txBlockRand;
 
-    difficulty
-        = m_mediator.m_dsBlockChain.GetLastBlock()
-              .GetHeader()
-              .GetDifficulty(); // TODO: Need to get the latest blocknum, diff, rand1, rand2
     // Verify nonce
     block_num
         = m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum()
         + 1;
     LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
               "dsblock_num            = " << block_num);
 
-    difficulty = POW_DIFFICULTY;
+    // Non-gensis block
     if (block_num > 1)
     {
-        difficulty
-            = m_mediator.m_dsBlockChain.GetLastBlock()
-                  .GetHeader()
-                  .GetDifficulty(); // TODO: Need to get the latest blocknum, diff, rand1, rand2
+        uint8_t expectedDSDiff = m_mediator.m_dsBlockChain.GetLastBlock()
+                                     .GetHeader()
+                                     .GetDSDifficulty();
+        uint8_t expectedDiff = m_mediator.m_dsBlockChain.GetLastBlock()
+                                   .GetHeader()
+                                   .GetDifficulty();
+
+        if (difficultyLevel != expectedDSDiff
+            && difficultyLevel != expectedDiff)
+        {
+            LOG_GENERAL(WARNING,
+                        "Difficulty level is invalid. difficultyLevel: "
+                            << to_string(difficultyLevel)
+                            << " Expected: " << to_string(expectedDSDiff)
+                            << " or " << to_string(expectedDiff));
+
+            // TODO: penalise sender in reputation manager
+            return false;
+        }
+    }
+    else if (block_num == 1)
+    {
+        if (difficultyLevel != POW_DIFFICULTY
```

Comment:
What is this check for?

Reply:
there is two difficulty level. To ensure ds committee don't waste time verifying on the difficulty, the node will declare the difficulty used. If it matched (strictly equal) either ds diff or diff, the ds committee will need to proceed to verify at the respective diff level